File: snippets/dart_snippets/lib/webhook.dart
import 'package:dart_snippets/sdk_instance.dart';

Future<void> registerWebhook() async {
  // ANCHOR: register-webook
  String webhookUrl = "https://your-nds-service.com/api/v1/notify?platform=ios&token=<PUSH_TOKEN>";
  await breezSDKLiquid.instance!.registerWebhook(webhookUrl: webhookUrl);
  // ANCHOR_END: register-webook
}

Future<void> unregisterWebhook() async {
  // ANCHOR: unregister-webook
  await breezSDKLiquid.instance!.unregisterWebhook();
  // ANCHOR_END: unregister-webook
}

File: snippets/react-native/webhook.ts
import { BindingLiquidSdk } from '@breeztech/breez-sdk-liquid-react-native'

const _registerWebhook = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: register-webook
  try {
    sdk.registerWebhook('https://your-nds-service.com/api/v1/notify?platform=ios&token=<PUSH_TOKEN>')
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: register-webook
}

const _unregisterWebhook = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: unregister-webook
  try {
    sdk.unregisterWebhook()
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: unregister-webook
}
